include/onnxruntime/core/framework/tensor.h

```diff
@@ -77,6 +79,20 @@ class Tensor final {
    */
   Tensor(MLDataType p_type, const TensorShape& shape, std::shared_ptr<IAllocator> allocator);
 
+  /**
+   * Create tensor with given type, shape, pre-allocated memory and allocator which will be used to free the pre-allocated memory.
+   * This function won't check if the preallocated buffer(p_data) has enough room for the shape.
+   * However, this function will de-allocate the buffer upon the tensor getting destructed.
+   * \param p_type Data type of the tensor
+   * \param shape Shape of the tensor
+   * \param p_data A preallocated buffer. Can be NULL if the shape is empty.
+   *              Tensor does not own the data and will not delete it
+   * \param deleter Allocator used to free the pre-allocated memory
+   * \param offset Offset in bytes to start of Tensor within p_data. 
+   */
+  Tensor(MLDataType p_type, const TensorShape& shape, void* p_data, std::shared_ptr<IAllocator> deleter,
```

Comment:
introduced a new ctor for the Tensor class so that it can now take in a pre-allocated buffer AND a deleter. I felt this is cleaner than modifying the exsiting ctor that takes in a pre-allocated buffer and a memory info to additionally take in an optional deleter.